motor_control/throttle/Button.ino

```diff
@@ -0,0 +1,39 @@
+#include "Button.h"
+
+Button::Button(int ft = 5):Throttle(){
+    isPressed = false;
+    fullThrottle = ft;
+}
+
+int Button::mappingFunction(int f, int s){
+  //linear
+  if(0){
+    return min(s/fullThrottle, 1) * 1023;
+  }
+  //exponential
+  if(1){
+  }
+  //logarithmic
+  if(2){
+  }
+}
+
+int Button::readThrottle() {
+  /*
+   * Reads the throttle input from analog signal,
+   * Returns 0~1023
+   */
+    if(throttleIn == HIGH){
```

Comment:
`throttleIn` is not the name of the raw value that the base class implements. You're looking for `m_raw_value`.